src/pages/Signup.jsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth } from "../contexts";
+
+function Signup() {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [pass, setPass] = useState("");
+
+  const { userSignup, setUserData } = useAuth();
+  const navigate = useNavigate();
+
+  return (
+    <div className="center-hv">
+      <div className="auth-form">
+        <form
+          onSubmit={(e) =>
+            userSignup(e, email, pass, name, setUserData, navigate)
+          }
+        >
+          <h2>Sign up</h2>
+          <div className="inp-unit">
+            <label htmlFor="fname">Name</label>
+            <input
+              type="text"
+              name="fname"
+              placeholder="johnsangram"
+              onChange={(e) => setName(e.target.value)}
+            />
+          </div>
+          <div className="inp-unit">
+            <label htmlFor="email">Email</label>
+            <input
+              type="email"
+              name="email"
+              placeholder="johnsangram@gkool.com"
+              onChange={(e) => setEmail(e.target.value)}
+            />
+          </div>
+          <div className="inp-unit">
+            <label htmlFor="password">Password</label>
+            <input
+              type="password"
+              name="email"
+              placeholder="**********"
+              onChange={(e) => setPass(e.target.value)}
```

Comment:
Just a sugggestion- you can make these inputs controlled components by setting value as the state.

Reply:
Okay Surbhi, I'll look into it :+1: